### ISSUE TITLE
Fix local variables in matchers

### ISSUE THREAD

**Issue by NilsHaldenwang** [bakkdoor/fancy#46]
```
match "foobarbaz" -> {
  case r{foo([a-z]+)baz} -> |matcher|
    foo = 10
    matcher[1] println  # => bar
} 
```

foo = 10 makes locals in match clause wrong atm.

**Comment by vic** [comment]
Guess we could fix this by always creating a block using the expression-list at the right of the -> , and if something matches call the generated block, now you've added support for having the match result like in |matcher| , we could use that as block arguments when specified.

**Comment by bakkdoor** [comment]
Yeah, that's what I thought of as well. But that might be a little overhead. We really just need to set a local, if the arg is given, that's it. Will look into it.

**Comment by bakkdoor** [comment]
Fixed bug for setting the correct local to the result of a match operation, if a matcharg is given in a case expression. Also added a spec for testing that.

Description: We need to use match_arg.name to get the identifier's symbol value, since LocalVariableAccess nodes (as any other) do the same. This ensures, that the identifier's local slot is set to the one used in the body as well. Closed by 3e207222d9344ed9493c5a4e3d2be39d25bdb44f
